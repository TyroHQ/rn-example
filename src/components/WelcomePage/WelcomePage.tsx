import React from "react";
import { Text } from "react-native";
import styled from "styled-components/native";
import { Placeholder } from "../Placeholder";

const WelcomePageContainer = styled.View`
  background-color: lightgrey;
  padding: 10px;
`;

export class WelcomePage extends React.Component<{}> {
  static defaultProps: WelcomePage["props"];
  render() {
    return (
      <Placeholder>
        <Placeholder>Header bar</Placeholder>
        <Placeholder>
          Content Area - Vector component
          <Placeholder>Title text</Placeholder>
          <Placeholder>Header text</Placeholder>
        </Placeholder>
        <Placeholder>Rectangular button</Placeholder>
      </Placeholder>
    );
  }
}
