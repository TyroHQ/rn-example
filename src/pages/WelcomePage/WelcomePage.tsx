import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";
import { Placeholder } from "@src/components/Placeholder";

const WelcomePageContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class WelcomePage extends React.Component<{}> {
  static defaultProps: WelcomePage["props"];
  render() {
    return (
      <Placeholder>
        <Placeholder>Header bar</Placeholder>
        <Placeholder>
          Content Area - Vector component
          <Placeholder>Title text</Placeholder>
          <Placeholder>Header text</Placeholder>
        </Placeholder>
        <Placeholder>Rectangular button</Placeholder>
      </Placeholder>
    );
  }
}
