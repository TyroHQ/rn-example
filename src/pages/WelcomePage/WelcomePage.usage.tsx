import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { WelcomePage } from "./";

export const basic = () => <WelcomePage />;
