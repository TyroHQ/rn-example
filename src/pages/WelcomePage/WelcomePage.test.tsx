import { shallow } from "enzyme";
import * as usage from "./WelcomePage.usage";

describe("<WelcomePage />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
