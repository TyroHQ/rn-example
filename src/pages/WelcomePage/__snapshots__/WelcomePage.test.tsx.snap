// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomePage /> renders well in basic mode 1`] = `
<Placeholder
  orientation="row"
>
  <Placeholder
    orientation="row"
  >
    Header bar
  </Placeholder>
  <Placeholder
    orientation="row"
  >
    Content Area - Vector component
    <Placeholder
      orientation="row"
    >
      Title text
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Header text
    </Placeholder>
  </Placeholder>
  <Placeholder
    orientation="row"
  >
    Rectangular button
  </Placeholder>
</Placeholder>
`;
